RUFAS/biophysical/animal/nutrients/nutrition_supply_calculator.py

```diff
@@ -0,0 +1,651 @@
+
+        Returns
+        -------
+        float
+            Total digestible calcium supply in the ration (g).
+
+        References
+        ----------
+        Animal Scientific Documentation [A.Cow.E.16]-[A.Heifer.E.16]
+
+        """
+        calcium_digestibility: dict[RUFAS_ID, float] = {}
+
+        for feed in feeds:
+            if feed.info.feed_type is FeedComponentType.FORAGE:
+                ca_digestibility = 0.3
+            elif feed.info.feed_type is FeedComponentType.CONC:
+                ca_digestibility = 0.6
+            elif feed.info.feed_type is FeedComponentType.MINERAL:
+                ca_digestibility = 0.95
+            else:
+                ca_digestibility = 0.0
+            calcium_digestibility[feed.info.rufas_id] = ca_digestibility
+
+        total = sum(
+            [
+                feed.amount
+                * calcium_digestibility[feed.info.rufas_id]
+                * feed.info.calcium
+                * GeneralConstants.PERCENTAGE_TO_FRACTION
+                for feed in feeds
+            ]
+        )
+
+        return total * GeneralConstants.KG_TO_GRAMS
+
+    @classmethod
+    def _calculate_phosphorus_supply(cls, feeds: list[FeedInRation]) -> float:
+        """
+        Calculates the phosphorus supply in the ration (kg).
+
+        Parameters
+        ----------
+        feeds : list[FeedInRation]
+            List of feeds in ration, including the amount and nutritive properties.
+
+        Returns
+        -------
+        float
+            Total digestible phosphorus supply in the ration (g).
+
+        References
+        ----------
+        Animal Scientific Documentation [A.Cow.E.17]-[A.Heifer.E.17]
+
+        """
+        phosphorus_digestibility: dict[RUFAS_ID, float] = {}
+
+        for feed in feeds:
+            if feed.info.feed_type is FeedComponentType.FORAGE:
+                p_digestibility = 0.64
+            elif feed.info.feed_type is FeedComponentType.CONC:
+                p_digestibility = 0.7
+            elif feed.info.feed_type is FeedComponentType.MINERAL:
+                p_digestibility = 0.8
+            else:
+                p_digestibility = 0.0
+            phosphorus_digestibility[feed.info.rufas_id] = p_digestibility
+
+        total = sum(
+            [
+                feed.amount
+                * phosphorus_digestibility[feed.info.rufas_id]
+                * feed.info.phosphorus
+                * GeneralConstants.PERCENTAGE_TO_FRACTION
+                for feed in feeds
+            ]
+        )
+
+        return total * GeneralConstants.KG_TO_GRAMS
+
+    @classmethod
+    def _calculate_metabolizable_protein_supply(
+        cls,
+        feeds: list[FeedInRation],
+        dry_matter_intake: float,
+        actual_tdn_percentages: dict[RUFAS_ID, float],
+        body_weight: float,
+    ) -> float:
+        """
+        Calculates amount of metabolizable protein in ration (kg).
```

Comment:
```suggestion
        Calculates amount of metabolizable protein in ration.
```
Not sure about adding this to the function description. I think if you're concerned about noting that the inputs are in kg and the output is in g, it should stay in the params and returns section (just my opinion).